src/pkg/path/resource_path.go

```diff
@@ -253,21 +253,29 @@ func (rp dataLayerResourcePath) Dir() (Path, error) {
 }
 
 func (rp dataLayerResourcePath) Append(
-	element string,
 	isItem bool,
+	elems ...string,
 ) (Path, error) {
 	if rp.hasItem {
 		return nil, clues.New("appending to an item path")
 	}
 
 	return &dataLayerResourcePath{
-		Builder:  *rp.Builder.Append(element),
+		Builder:  *rp.Builder.Append(elems...),
 		service:  rp.service,
 		category: rp.category,
 		hasItem:  isItem,
 	}, nil
 }
 
+func (rp dataLayerResourcePath) AppendItem(item string) (Path, error) {
+	if rp.hasItem {
+		return nil, clues.New("appending to an item path")
+	}
```

Comment:
do we need to check for this here and return an error? It seems like it would be fine to just rely on the call below to catch this and then we don't have to worry about the error messages drifting apart. There should also a test to make sure this fails as expected so changes down the road should fail the test

Reply:
fair, fair.  Test added.